src/python/pants/engine/target.py

```diff
@@ -190,3 +227,18 @@ def has_fields(self, fields: Iterable[Type[Field]]) -> bool:
             if self._find_registered_field_subclass(unrecognized_field) is None:
                 return False
         return True
+
+
+class BoolField(PrimitiveField):
```

Comment:
+1 to @illicitonion  and +1 to dataclasses! This substantially reduces boilerplate.